code/lib/preview-api/src/modules/preview-web/PreviewWeb.test.ts

```diff
@@ -2735,6 +2735,60 @@ describe('PreviewWeb', () => {
       });
     });
 
+    describe('if called twice simultaneously', () => {
+      it('does not get renders confused', async () => {
+        const [importGate, openImportGate] = createGate();
+        const [importedGate, openImportedGate] = createGate();
+        const newImportFn = vi.fn(async (path) => {
+          openImportedGate();
+          await importGate;
+          return importFn(path);
+        });
+
+        document.location.search = '?id=component-one--a';
+        const preview = await createAndRenderPreview();
+        mockChannel.emit.mockClear();
+
+        preview.onStoriesChanged({ importFn: newImportFn });
+        await importedGate;
+        preview.onStoriesChanged({ importFn });
+
+        openImportGate();
+        await waitForRender();
+
+        expect(preview.storyRenders.length).toEqual(1);
```

Comment:
I don't love reaching into the internals of `preview` here. An alternative would be to try and reproduce the issue via making an args change and checking for `storyArgsUpdated`. WDYT?

Reply:
I think the alternative would be a higher-level test which could regress for multiple reasons. I like the focus of this test.